code/modules/atmospherics/machinery/atmosmachinery.dm

```diff
@@ -301,7 +301,9 @@ Pipelines + Other Objects -> Pipe network
 #define VENT_SOUND_DELAY 30
 
 /obj/machinery/atmospherics/relaymove(mob/living/user, direction)
-	if(!(direction & initialize_directions)) //cant go this way.
+	direction = direction & initialize_directions
+	var/static/list/text_cardinals = list(TEXT_NORTH=TRUE, TEXT_EAST=TRUE, TEXT_SOUTH=TRUE, TEXT_WEST=TRUE)
+	if(!direction || !text_cardinals["[direction]"]) //cant go this way.

```

Comment:
I really don't think this tiny optimization is necessary. Just do this:
```
direction &= initialize_directions
if(!(direction in GLOB.cardinals))
```

Reply:
using an associative list like that is v ugly imo and ventcrawling is not extremely performance sensitive

Reply:
I didn't do `&=` because I was offhandedly thinking about the bug related to it but it doesn't matter here.

As for performance: eh whatever I guess. I've gotten too into habit to avoid `in`